### DISCUSSION TITLE
Adapter with custom schema? (combine user and account)

### DISCUSSION THREAD

**Discussion by zimonitrome** [nextauthjs/next-auth#7744] [Help]
### Question 💬

Hello, I am using the [Prisma adapter for authjs](https://authjs.dev/reference/adapter/prisma) (semi relevant).

The schema defines three tables: `Account`, `Session`, `User`, `VerificationToken`.

In my app, I want users to be able to login with multiple accounts even if they have the same `email`. Thus it makes more sense to use a single table that the combination of `User` and `Account` (say `UserAccount`) that is unique on `provider` and `providerAccountId`. This would work, but if I want to change the [adapter code](https://github.com/nextauthjs/next-auth/blob/main/packages/adapter-prisma/src/index.ts#L221-L278) I can't get the necessary data to create a new `UserAccount`, since `createUser` and `linkAccount` are mandatory and give partial information to each function.

Is there a way to get around this?

### How to reproduce ☕️

Follow the page for Prisma adapter: https://authjs.dev/reference/adapter/prisma

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by zimonitrome** [comment]
In the meantime I'm trying to get around this issue. When I login with an account which email is already in my database but through another provider I get a [callbackrouteerror](https://authjs.dev/reference/core/errors/#callbackrouteerror) but the [ErrorPage](https://next-auth.js.org/configuration/pages#error-page) only returns this as `ErrorType` `"default"`. The URL query string says `?error=CallbackRouteError` but I don't think this is specific enough.

Preferably I would like to inform the user that they `email` is already occupied.

**Comment by mikalai-s** [comment]
I have the same issue, which can be narrowed down to the simple fact that it is currently impossible to authenticate through different authentication providers with the same email using next-auth. I believe the issue can be resolved by simply adding a provider field to the user table. This way, when checking for an existing user [here](https://github.com/nextauthjs/next-auth/blob/c7dec376a1272a0877c7e6fb2fc4173f08369f9f/packages/core/src/lib/callback-handler.ts#L191), we would check for the existence of the `email` + `provider` pair.
